sigle/src/modules/editor/extensions/Twitter/index.ts

```diff
@@ -0,0 +1,100 @@
+import { Node, mergeAttributes, nodePasteRule } from '@tiptap/core';
+import { ReactNodeViewRenderer } from '@tiptap/react';
+import { TwitterComponent } from './TwitterComponent';
+import { getTweetIdFromUrl, TWITTER_REGEX_GLOBAL } from './utils';
+
+// TODO handle global paste
+
+declare module '@tiptap/core' {
+  interface Commands<ReturnType> {
+    twitter: {
+      /**
+       * Insert a video embed
+       */
+      setTweet: () => ReturnType;
+    };
+  }
+}
+
+export const Twitter = Node.create({
+  name: 'twitter',
+  group: 'block',
+  selectable: true,
+  draggable: true,
+
+  addOptions() {
+    return {
+      addPasteHandler: true,
+    };
+  },
+
+  addAttributes() {
+    return {
+      ['data-twitter-id']: {
+        default: null,
+        // force correct id
+        parseHTML: (element) => element.getAttribute('data-twitter-id'),
+      },
+      url: {
+        default: null,
+      },
+      pasted: false,
+    };
+  },
+
+  addCommands() {
+    return {
+      setTweet:
+        () =>
+        ({ commands }) => {
+          commands.insertContent({
+            type: this.name,
+          });
+          commands.focus();
```

Comment:
is this doing anything?

Reply:
guess we don't need this as we have the solution with setTimout - removed 👍